### PR TITLE
Bugfix/snake camel dash

### PR DESCRIPTION
**Pull request by MarlonPassos-git**
## Description
Please provide a detailed description of the changes and the intent behind them :)

## Checklist
- [x] Changes are covered by tests if behavior has been changed or added
- [x] Tests have 100% coverage
- [x] The version in `package.json` has been bumped according to the changes made and standard semantic versioning rules

## Resolves
#96

[I saw an interesting thread about things that are not 100% compared to lodash](https://github.com/rayepps/radash/discussions/96). I made this pr to fix these "bug"

### CODE REVIEW

File: src/string.ts [hunk 1]
```diff
@@ -15,9 +15,13 @@ export const capitalize = (str: string): string => {
  *
  * camel('hello world')   -> 'helloWorld'
  * camel('va va-VOOM') -> 'vaVaVoom'
+ * camel('helloWorld') -> 'helloWorld'
  */
 export const camel = (str: string): string => {
-  const parts = str?.split(/[\.\-\s_]/).map(x => x.toLowerCase()) ?? []
+  const parts = str
+    ?.replace(/([A-Z])+/g, capitalize)
```
Comment (rayepps) [inline src/string.ts:22]: I did not know you could use a function as the replacement argument 🤯 ❤️

Comment (MarlonPassos-git) [reply src/string.ts:22]: yes, that's pretty cool, in my service we use it this way with maps, filters, etc... It makes the code more readable

**Review by rayepps — APPROVED** [review]
Thanks @MarlonPassos-git so much for this PR, I like the simple fix and how consistent it is with the existing code. I also ❤️ the added tests!

**Comment by rayepps** [comment]
This PR has been published 🚀 to npm in version `8.0.3`

**Comment by wouter-leistra** [comment]
> Thanks @MarlonPassos-git so much for this PR, I like the simple fix and how consistent it is with the existing code. I also ❤️ the added tests!

Thanks so much for the fix! 

Just want to point out that the usage of toLowercase may not function as expected when some other locale is used as input. I believe one should prefer the usage of to LocaleLowerCase to get 100% accuracy, but as the two functions mostly produce the same results I'm not sure if you feel it is worth the change and possibly additional optional parameter.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLocaleLowerCase

**Comment by rayepps** [comment]
@wouter-leistra thanks for pointing this out, I've never had a need to work with i8 before so I've never used locale.

I'm down to add it or would accept a pr.

**Pull request merged**